File: Calculator.API/Models/ResponseBase.cs
using System.Collections.Generic;

namespace Calculator.API.Models
{
    public class ResponseBase
    {
        public decimal EndBalance { get; set; }
        public decimal StartPrincipal { get; set; }
        public decimal TotalContribution { get; set; }
        public decimal TotalInterest { get; set; }
        public List<InterestSchedule> InterestSchedule { get; set; }
    }
}

File: CalculatorUI/src/app/layout/shared/models/responseBase.ts
import { InterestSchedule } from "./InterestSchedule";

export class ResponseBase{
    endBalance!: number;
    startPrincipal!: number;
    totalContribution!: number;
    totalInterest!: number;
    inflationAdjustment!: number;
    totalTax!: number;
    interestSchedule: InterestSchedule[] = [];
}